### ISSUE TITLE
Can't transfer water from vehicle faucet.

### ISSUE THREAD

**Issue by erathia65** [CleverRaven/Cataclysm-DDA#49197]
# Describe the bug

 DEBUG    : error in activity data: "could not find source vehicle for liquid transfer"

 FUNCTION : void activity_handlers::fill_liquid_do_turn(player_activity*, player*)
 FILE     : src/activity_handlers.cpp
 LINE     : 1609

# Steps To Reproduce

Spawn a vehicle
Install a faucet
Install a tank
Add water to tank
Try to fill a container with water
???
profit.

# Expected behavior

Won't transfer your liquid into a container.

# Versions and configuration

Latest experimental

**Comment by Anaglyphic** [comment]
_**"could not find source vehicle"**_ is your issue, not the tank nor the spigot; please change the subject line. and hey! stop cheating with debug!  Seriously though, did you test anything before you came to report this? Did you save, exit, relaunch? It will probably work when the world data is re-parsed from you making said vehicle just _<pop!>_ into existence.

If you spawned instead of a whole vehicle, just some steel frames, a water tank (60L), a faucet, then assembled them, filled them, and used them?

```
┌────────────────────────────────┐
│ Select an action               │
├────────────────────────────────┤
│ e Examine vehicle              │
│ K Remember vehicle position    │
│ c Fill a container with water  │
│ d Have a drink                 │
└────────────────────────────────┘
```
Everything works as expected, containers carried and nearby can be filled.

**Comment by erathia65** [comment]
So first, debug is meant to debug right, if i encounter a bug during playthrough that does not go away after restart, the debug menu is there to help me reproduce the bug, to help DEBUG it right?

And second, the bug appeared during a normal playthrough, i did not spawn a vehicle during my playthrough, here are some screenshots.
![image](https://user-images.githubusercontent.com/3441742/120956434-d2eea400-c753-11eb-9aa5-f6904e4d6eed.png)
![image](https://user-images.githubusercontent.com/3441742/120956452-dbdf7580-c753-11eb-98c1-7f0a968dd211.png)
![image](https://user-images.githubusercontent.com/3441742/120956462-e26ded00-c753-11eb-93ab-af5bd231bac6.png)
![image](https://user-images.githubusercontent.com/3441742/120956472-e8fc6480-c753-11eb-8aae-dff7e901e642.png)
![image](https://user-images.githubusercontent.com/3441742/120956483-ef8adc00-c753-11eb-8658-9d031ebae769.png)
![image](https://user-images.githubusercontent.com/3441742/120956490-f44f9000-c753-11eb-9d96-8bd12745e3e5.png)

**Comment by erathia65** [comment]
I can have a drink, that's not the problem, i cant pour water in any container, carried or around. Like my issue said.
Reproductible on clean game.

**Comment by Anaglyphic** [comment]
Sorry, I was attempting to be funny, you know, sarcasm. Everyone uses debug to test things. Also to just play. Obviously if debug had caused it, it's not the original problem - debug should just "speed up" the processes as it were. But normally vehicles don't just pop into existence was my point, I wanted to see if that was the issue, you say it isn't. Great, now we know! But I was just (attempting) to be funny, honestly. 

But you also need to turn debug itself on, not just use the debug menu to give items or recreate scenarios, yeah? An error message showing what line in the code doesn't help, we already know that with just grep or any editor.

Luckily that text screenshot does gives us more, what the hell is a "CRIT canteen"? It's not part of the base game, it's a mod, and you didn't mention any mods in your initial post, this is why filling out the issue tracker as requested is important. Could you try a regular base game container like a ubiquitous plastic bottle, and not something from Soupster89's C.R.I.T. Expansion Mod? That will confirm it's it's a mod incompatibility and you should make him aware of this thread.

PS: I got to play UltiCa for a few months. Not that you care, but wow, it's like looking at a different game. It's funny how windows people only play the "grahpic-y" tilesets and the linux people only play ascii or "ascii-y" tilesets... I literally don't know what I'm looking at.

**Comment by erathia65** [comment]
Oh yeah it's a canteen with a thermonuclear generator inside to heat up the liquid when in need, but i didn't specify it because i tried to reproduce it in debug, with only the vanilla game, no C.R.I.T. Expansion, no Magiclysm, no aftershock, just vanilla DDA, and my screen are from my main gameplay, therefore the unusual items, here's some more screenshots with vanilla items.
![image](https://user-images.githubusercontent.com/3441742/121038738-cd737700-c7b0-11eb-93fa-50d0d8a99780.png)
![image](https://user-images.githubusercontent.com/3441742/121038752-d2382b00-c7b0-11eb-92e7-ff61698a934e.png)

**Comment by NetSysFire** [comment]
appears to be fixed, see #54642. please re-open if you can still reproduce this in the latest experimental.
